app/controllers/checkouts_controller.rb

```diff
@@ -7,4 +7,12 @@ def create
 
     redirect_to root_path
   end
+
+  def destroy
+    @book = Book.find(params[:book_id])
+    @checkout = Checkout.where(user: current_user, book: @book)
+    @checkout.destroy_all
```

Comment:
Actually, thinking about this some more -- does it really matter which user returned the book? If I think about the way my library works, I don't identify myself when returning books. Each book is uniquely identified and when someone (anyone) returns it, the checkout is removed/ended.

I wonder if we could use methods on the models to encapsulate the behavior we want here. A few options I could see:

```ruby
Book.find(params[:book_id]).return
Checkout.find(params[:book_id]).return
Return.process(book)
```

The first would feel a little weird because it would require that the `Book` model know how to handle, for example, deleting or updating a checkout instance.

The second seems okay. It's not exactly natural language to say "return this checkout". From an OO perspective it seems like the right object to receive some sort of message and handle this though.

The third option is a plain ruby object (not an ActiveRecord instance) that just knows how to process returns. It's probably premature to go to something like this right now, but it would be a good option if returns became a multi-step process at some point. Something to keep in mind.

In any case, we're encoding the knowledge that returning a book means destroying the checkout record in a method. If at some point we decide "no, we just want to have a `returned_at` column rather than deleting the record` we could update the model and not the controller or anywhere else where returns may happen.

The code you already have here is simple enough that all of those options may be overkill at this point, but I thought it at least instructive to consider it and at the very least keep it in mind as an option as we move forward if checkouts get more complicated, for instance.

What do you think?